rust/src/digest.rs

```diff
@@ -263,10 +265,11 @@ impl Algorithm {
 
 /// Check if we do SHA3, which is not in the agile API and hence has to be
 /// handled differently.
-const fn is_sha3(alg: Algorithm) -> bool {
+/// SHA3 256 is supported.
+const fn is_unsupported_sha3(alg: Algorithm) -> bool {
     matches!(
```

Comment:
This, I don't understand. The comment says "SHA3 256 is supported" but we only return true on `Sha3_{224,384,512}`. Shouldn't we use `Sha3_256` in the match pattern only?

Reply:
Dough. Could you refactor that to either `is_supported_sha3` or rename to `is_not_supported_sha3`? The `un` (mixed with the last line of the comment) makes this easy to overlook (as demonstrated) :P

Reply:
I renamed to `does_not_support_streaming`

Reply:
Thanks 🙏